Modules/Core/Common/include/itkConstNeighborhoodIterator.hxx

```diff
@@ -210,9 +210,8 @@ template <typename TImage, typename TBoundaryCondition>
 ConstNeighborhoodIterator<TImage, TBoundaryCondition>::ConstNeighborhoodIterator()
 
 {
-  IndexType zeroIndex;
-  zeroIndex.Fill(0);
-  SizeType zeroSize;
+  constexpr IndexType zeroIndex = { { 0 } };
+  SizeType            zeroSize;
```

Comment:
Is a similar PR for `Size`s in the works? There should be less of those than zero indices. 😄

Reply:
> Is a similar PR for `Size`s in the works? There should be less of those than zero indices.

@dzenanz  Well, I can surely take a look at those as well. But first I'd like to see if this `Index` PR can make it onto the master... Wow, three approvals already  😄 

Indeed, zero-`Size`s appear far less popular than zero-`Index`es 😄😄

Reply:
I'd really like to see:
```c++
constexpr RegionType                  region( {{0}}, {{128}} );

constexpr RegionType3D                  region( {{0}}, { {128, 256, 384} } );
```

:). if someone else wants to put the work in.

Reply:
@hjmjohnson Thanks for merging! You wrote:

>     constexpr RegionType3D region( {{0}}, { {128, 256, 384} } );

Nice suggestion! However, an object of type `itk::ImageRegion<N>`  cannot be constexpr, because it has virtual member functions!  For performance reasons, it might be worth considering to remove those virtual member functions. Eventually... but that may take some more time.

Now I just realized that  `itk::ImageRegion<N>` has another constructor, which only has a `Size` parameter (zero-initializing the Index automatically). So instead, we could write:

     const RegionType3D region( { {128, 256, 384} } );

Would you like that too?

Reply:
Yes. 

Good point on constexpr.  const is what I should have requested.

Reply:
If it were a pointer `itk::ImageRegion<N> *` I could imagine why virtual member functions might not be known until runtime, but since it is not a pointer, what's the issue with virtual member functions, especially if those member functions aren't used in the constructor?  I need to go back to the textbooks....